### PR TITLE
[DOCS] Adds data frame API response codes for allow_no_match

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/pull/43490

This PR adds a response codes section to the documentation for the get data frame transforms, get data frame transform statistics, and stop data frame transforms APIs. It also edits the description of the allow_no_match query parameter.

### CODE REVIEW

File: docs/reference/data-frames/apis/get-transform.asciidoc [hunk 1]
```diff
@@ -51,17 +51,20 @@ see {stack-ov}/security-privileges.html[Security privileges] and
 ==== {api-query-parms-title}
 
 `allow_no_match` (Optional)::
-  (boolean) Whether to ignore if a wildcard expression matches no
-  {dataframe-transforms}. This includes `_all` string or when no transforms have
-  been specified. The default is `true`.
+  (boolean) Specifies what to do when the request contains a wildcard expression
+  and there are no {dataframe-transforms} that match. The default value is
+  `true`, which returns an empty `transforms` array in this scenario. If it's
```
Comment (benwtrent) [inline docs/reference/data-frames/apis/get-transform.asciidoc]: It is also if there is a partial match. If somebody supplies the patter `test-id*,missing*` and `missing*` does not match but `test-id*` does, we will return the `test-id*` matched transforms without the `missing*`. 

If the `allow_no_match` is `false`, we would through the 404, even if there is a partial match for one of the patterns.

File: docs/reference/data-frames/apis/stop-transform.asciidoc [hunk 2]
```diff
@@ -72,6 +75,14 @@ All {dataframe-transforms} can be stopped by using `_all` or `*` as the
   completely stops. If set to `false`, the API returns immediately and the
   indexer will be stopped asynchronously in the background. Defaults to `false`.
 
+[discrete]
+[[stop-data-frame-transform-response-codes]]
+==== {api-response-codes-title}
+
+`404`::
+  If `allow_no_match` is `false`, this code indicates that there are no
+  resources that match the request.
```
Comment (benwtrent) [inline docs/reference/data-frames/apis/stop-transform.asciidoc]: This should refer to `missing resources` a partial match is also errored out in this case.

**Comment by lcawl** [comment]
Thanks for the feedback, @benwtrent. I've updated the three pages to try to be clearer about the partial match behaviour.

**Review by benwtrent — APPROVED** [review]
📝 ✔️

**Comment by lcawl** [comment]
@elasticmachine update branch

**Pull request merged**